managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -117,6 +119,9 @@ public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
                 return;
             }
             updateReadPosition(nexReadPosition);
+            if (lostLedger != null) {
+                clearIncompleteAckedRecordsFromLedger(lostLedger);
```

Comment:
Call `cursor.getManagedLedger().removeNonRecoverableLedger(ledgerId);` directly to avoid adding a useless stack frame.

Reply:
fixed